src/apps/DesktopAppPage.vala

```diff
@@ -245,14 +376,37 @@ namespace Ilia {
                 model.get_value(iter, ITEM_VIEW_COLUMN_NAME, out app_info);
                 strval = app_info.get_string();
 
-                if (strval != null && strval.down().contains(query_string)) return true;
+                // Check if this is a favorite app (will be used for sorting)
+                model.get_value(iter, ITEM_VIEW_COLUMN_IS_FAVORITE, out app_info);
+                bool is_favorite = app_info.get_boolean();
+
+                // Exact substring match - highest priority
+                if (strval != null && strval.down().contains(query_string)) {
+                    return true;
+                }
+
+                //fuzzy matching on the app name
+                if (strval != null && fuzzy_match(strval, query_string, 50)) {
+                    return true;
+                }
 
+                // Check in keywords as well
                 model.get_value(iter, ITEM_VIEW_COLUMN_KEYWORDS, out app_info);
                 strval = app_info.get_string();
 
-                return strval != null && strval.down().contains(query_string);
+                // Try exact substring match on keywords
+                if (strval != null && strval.down().contains(query_string)) {
+                    return true;
+                }
+
+                // Try fuzzy matching on keywords with a lower threshold
+                if (strval != null && fuzzy_match(strval, query_string, 40)) {
+                    return true;
+                }
+
+                return false;
             } else {
-                return true;
+                return true; // Show all items when no query
```

Comment:
nit: Move this to the top of the function with the guard condition:
```vala
if (strval == null) return true
```

Then you can remove the `strval != null`  condition from all the branches to make it more readable.